app/views/admin/custom_fields/form/_search_filter.haml

```diff
@@ -4,13 +4,13 @@
       - custom_field.with_type do |type|
         - case type
         - when :date_field
-          = form.check_box :search_filter, disabled: true
+          = form.check_box :search_filter, checked: false, disabled: true
```

Comment:
It just looks more logical that when text or date fields cannot be made a filter and the box is shown as disable the checkbox is not checked.